subsys/bluetooth/controller/hci_driver.c

```diff
@@ -244,36 +271,172 @@ void SIGNALLING_Handler(void)
 	k_sem_give(&sem_signal);
 }
 
-static int hci_driver_init(struct device *unused)
+u8_t bt_read_static_addr(bt_addr_le_t *addr)
 {
-	ARG_UNUSED(unused);
+	if (((NRF_FICR->DEVICEADDR[0] != US32_MAX) ||
+	     ((NRF_FICR->DEVICEADDR[1] & UINT16_MAX) != UINT16_MAX)) &&
+	    (NRF_FICR->DEVICEADDRTYPE & 0x01)) {
+		sys_put_le32(NRF_FICR->DEVICEADDR[0], &addr->a.val[0]);
+		sys_put_le16(NRF_FICR->DEVICEADDR[1], &addr->a.val[4]);
+
+		/* The FICR value is a just a random number, with no knowledge
+		 * of the Bluetooth Specification requirements for random
+		 * static addresses.
+		 */
+		BT_ADDR_SET_STATIC(&addr->a);
+
+		addr->type = BT_ADDR_LE_RANDOM;
+		return 1;
+	}
+	return 0;
+}
+
+static int ble_init(void)
+{
+	s32_t err = 0;
+
+	nrf_lf_clock_cfg_t clock_cfg;
 
-	u32_t err = blectlr_init(host_signal);
+#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC
+	clock_cfg.lf_clk_source = NRF_LF_CLOCK_SRC_RC;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_XTAL
+	clock_cfg.lf_clk_source = NRF_LF_CLOCK_SRC_XTAL;
+#elif CLOCK_CONTROL_NRF_K32SRC_SYNTH
+	clock_cfg.lf_clk_source = NRF_LF_CLOCK_SRC_SYNTH;
+#else
+#error "Clock source is not defined"
+#endif
 
-	if (err) {
-		/* Probably memory */
-		return -ENOMEM;
+#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_500PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_500_PPM;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_250PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_250_PPM;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_150PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_150_PPM;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_100PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_100_PPM;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_75PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_75_PPM;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_50PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_50_PPM;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_30PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_30_PPM;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_20PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_20_PPM;
+#elif CLOCK_CONTROL_NRF_K32SRC_10PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_10_PPM;
+#elif CLOCK_CONTROL_NRF_K32SRC_5PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_5_PPM;
+#elif CLOCK_CONTROL_NRF_K32SRC_2PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_2_PPM;
+#elif CLOCK_CONTROL_NRF_K32SRC_1PPM
+	clock_cfg.accuracy = NRF_LF_CLOCK_ACCURACY_1_PPM;
+#else
+#error "Clock accuracy is not defined"
+#endif
+	clock_cfg.rc_ctiv = BLE_CONTROLLER_RECOMMENDED_RC_CTIV;
+	clock_cfg.rc_temp_ctiv = BLE_CONTROLLER_RECOMMENDED_RC_TEMP_CTIV;
+
+	err = MULTITHREADING_LOCK_ACQUIRE();
+	if (!err) {
+		err = ble_controller_init(blectlr_assertion_handler,
+					  &clock_cfg,
+					  SWI5_IRQn);
+		MULTITHREADING_LOCK_RELEASE();
 	}
+	if (err < 0) {
+		return err;
+	}
+
+	ble_controller_cfg_t cfg;
+
+	cfg.master_count.count = 1;
```

Comment:
Can we improve the defaults here?
The values should configurable through Kconfig.
The defaults should be changed to be consistent with SoftDevice defaults:

DLE: 27 bytes, 3 packet buffers ( I think )
Central count 3 for s132/s140, 0 for s112
Periph count 1:
Event length: 2500 us

Reply:
We should, however not for this release. I will create an internal issue to follow this up for upcoming NCS releases. We should consider to align with pre-existing options in Zephyr and find what differences we have.